differ/mutationDiffer.go

```diff
@@ -1086,7 +1094,7 @@ func areGetMetaResultsTheSame(result1Raw, result2Raw interface{}) bool {
 	} else {
 		// Only compare json part of datatype
 		return result1.Cas == result2.Cas && result1.SeqNo == result2.SeqNo && result1.Flags == result2.Flags &&
-			result1.Expiry == result2.Expiry && result1.Deleted == result2.Deleted && (result1.Datatype&base.JSONDataType == result2.Datatype&base.JSONDataType)
+			result1.Expiry == result2.Expiry && result1.Deleted == result2.Deleted && (result1.Datatype == result2.Datatype)
```

Comment:
Why are we changing this? Do we know the original reason for keeping it JSON data type?

Reply:
I don't know the reason but logically it makes sense to prevent re-raising false positives for flags that we don't know about.

Wouldn't it be safer to compare:
1. If one has JSON flag set, the other should too
2. If one has Xattr flag set, the other should too

Ignore comparing of other flags.